### DISCUSSION TITLE
channel count thoughts...

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#566] [General]
Hi;
all the tests I ran until now had the default channel count which is 1
today I have increased the count to dynamically take the cores from my machine (per the example in the book store)
microstream, upon starting, 'complained' that there is a 'former' db file that was not created with the new channel count and exited...
this seems to me like a big deal:
-what happens to the DB data of a customer when I add a CPU on his machine for example or even just tune the channel count in a more advanced product version that I deploy? this will render the DB non functional...
is there a way to bypass this behavior?
your help is apprecated...

**Comment by hg-ms** [comment]
The channel count can’t be changed by a simple configuration change. To change the channel count of a storage the whole storage has to be “converted”.
Microstream version 8 provides a tool for that:
https://github.com/microstream-one/microstream/blob/master/storage/embedded-tools/storage-converter/src/main/java/one/microstream/storage/embedded/tools/storage/converter.

**Reply by guykatz** [reply]
thanks @hg-ms 
seems to me like this might be a bigger issue....
imagine a cloud env where you might increase resources from time to time...
I think maybe MS better detect a difference in channels and migrate automatically (assuming its a quick process)
or at least give this behavior as a config option we can turn on.
many times we do not control the HW.... 
for me, I cant use the dynamic channel allocation because of this at this moment...
maybe you can tag this as a feature suggestion or something.
thanks
